### PR TITLE
FM2-444: Add the ability to save Location Tags

### PR DESCRIPTION
**Pull request by mozzy11**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-444

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by mozzy11** [comment]
cc @pmanko

**Comment by ibacher** [comment]
@mozzy11 I think we need to reconsider this. In particular, I think the existing code translating location tags is... not right. It seems to me that the way this will work will be to create a tag per location created by FHIR. So, e.g., instead of ending up with one "mCSD Location" tag, we will end up with one per-location.

I also think we should break out the responsibilities into different parts

1. UUID on new items should be assigned at the service create method (actually ideally the one in the BaseFhirService... I thought we did that there already but apparently not).
2. We probably should have a translator for tags that will look up an existing tag (if any) and assign that to the location. I would not create a new tag while creating a new location... we should ignore non-existent tags.

File: test-data/src/test/resources/org/openmrs/module/fhir2/providers/LocationWebTest_create.json [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+  "meta": {
+    "versionId": "1",
+    "lastUpdated": "2021-11-30T08:43:54.105+00:00",
+    "source": "#DNbZg5doCf3wLAkB",
+    "profile": [
+      "http://ihe.net/fhir/StructureDefinition/IHE.mCSD.FacilityLocation"
+    ],
+    "tag": [
+      {
+        "system": "http://facility_registry/mcsd_location",
```
Comment (ibacher) [inline test-data/src/test/resources/org/openmrs/module/fhir2/providers/LocationWebTest_create.json:12]: It's best we don't include a system here... there's no meaningful system for this. Really, in this case, it should just be "code", I think...

**Review by ibacher — APPROVED** [review]
LGTM. One final thing though, could you add a similar integration test for the R3 version of the same provider?

**Review by ibacher — APPROVED** [review]

**Pull request merged**